### PR TITLE
feat(notifications): Implement PATCH /subscription V2 API

### LINKED ISSUE #3143: feat(notifications): Implement PATCH /subscription V2 API
# 🚀 Feature Request

### Relevant Package

 This feature request is for support-notifications.

### Description

Per https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/default/patch_subscription, implement PATCH /subscription V2 API.

### PR DESCRIPTION
**Pull request by FelixTing**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the new behavior?
Per https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-notifications/2.x#/default/patch_subscription, implement PATCH /subscription V2 API.

Close #3143

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

File: internal/support/notifications/v2/application/subscription.go [hunk 1]
```diff
@@ -128,3 +130,45 @@ func DeleteSubscriptionByName(name string, ctx context.Context, dic *di.Containe
...
+	}
+	if dto.Name != nil && *dto.Name != subscription.Name {
+		return errors.NewCommonEdgeX(errors.KindContractInvalid, fmt.Sprintf("subscription name '%s' not match the existing '%s' ", *dto.Name, subscription.Name), nil)
+	}
+
+	requests.ReplaceSubscriptionModelFieldsWithDTO(&subscription, dto)
+	_, edgeXerr := dbClient.SubscriptionByName(subscription.Name)
+	if edgeXerr != nil {
+		return errors.NewCommonEdgeX(errors.Kind(edgeXerr), fmt.Sprintf("subscription '%s' existence check failed", subscription.Name), edgeXerr)
+	}
```
Comment (lenny-intel) [inline internal/support/notifications/v2/application/subscription.go:160]: Why is this check needed? Doesn't the code above fail if the `name`  doesn't exist or match? and the Delete below will fail if those above don't.

File: internal/support/notifications/v2/controller/http/subscription_test.go [hunk 2]
```diff
@@ -93,6 +94,32 @@ func addSubscriptionRequestData() requests.AddSubscriptionRequest {
 	return testAddSubscriptionReq
 }
 
+func updateSubscriptionRequestData() requests.UpdateSubscriptionRequest {
+	testUUID := ExampleUUID
+	testName := testSubscriptionName
+	testDescription := testSubscriptionDescription
+	var testUpdateSubscriptionReq = requests.UpdateSubscriptionRequest{
```
Comment (lenny-intel) [inline internal/support/notifications/v2/controller/http/subscription_test.go:101]: Use new factory method

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: internal/support/notifications/v2/application/subscription.go (hunk 1 above)
Comment (FelixTing) [reply internal/support/notifications/v2/application/subscription.go]: Yep, you are right. Removed.

File: internal/support/notifications/v2/controller/http/subscription_test.go (hunk 2 above)
Comment (FelixTing) [reply internal/support/notifications/v2/controller/http/subscription_test.go]: Updated subscription_test to use new factory method.

**Review by lenny-intel — APPROVED** [review]
LGTM, does this still need to be in Draft?

**Review by lenny-intel — APPROVED** [review]

**Comment by FelixTing** [comment]
> LGTM, does this still need to be in Draft?
Thanks. Converted to Ready-for-Review.

**Pull request merged**